### PR TITLE
feat: enhance python bindings

### PR DESCRIPTION
**Pull request by asalzburger**
This PR enhances the python bindings and adds the python based `json` handling.

It adds first python based examples as well:

```sh
polygons_from_obj.py
surface_material_from_json.py
```

### CODE REVIEW

File: python/setup.sh.in [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# This file is part of the Acts project.
```
Comment (niermann999) [inline python/setup.sh.in:1]: Should these be updated to actsvg?

File: python/python/actsvg/actsvg_json.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        surface_material_accessor = surface_material["accessor"]
+        # Convert the grid
+        accessor_grid = surface_material_accessor["grid"]
+        psm.grid, index_data = read_grid(accessor_grid)
+
+        # indicate the grid type
+        global_casts = surface_material["global_casts"]
+        if global_casts == ["binZ", "binPhi"]:
+            psm.grid.type = proto.grid.grid_type.z_phi
+            psm.grid.reference_r = 100
```
Comment (niermann999) [inline python/python/actsvg/actsvg_json.py:54]: Should this really be hardcoded?

**Review by niermann999 — COMMENTED** [review]
Looks good to me in general

File: python/src/io.cpp [hunk 3]
```diff
@@ -31,138 +31,153 @@ using surface = proto::surface<point3_collection>;
 void add_io_module(context& ctx) {
 
     auto& m = ctx.get("main");
+    // The io module: acts.io
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply python/src/io.cpp:34]: no `actsvg` of course

File: python/setup.sh.in (hunk 1 above)
Comment (asalzburger) [reply python/setup.sh.in:1]: No that's fine, actsvg is a subproject of Acts, so we can leave it, I suppose.

File: python/python/actsvg/actsvg_json.py (hunk 2 above)
Comment (asalzburger) [reply python/python/actsvg/actsvg_json.py:54]: No, this should be read - will be fixed when 

https://github.com/acts-project/acts/pull/3039

is merged, and the output format updated.

File: python/src/io.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (asalzburger) [reply python/src/io.cpp:34]: Fixed.

File: python/src/proto.cpp [hunk 4]
```diff
@@ -39,19 +39,116 @@ void add_proto_module(context& ctx) {
...
+        /// @param pcs is the point3 collection
+        /// @param f is the fill style
+        /// @param s is the stroke style
+        p.def(
+            "create_polygon",
+            [](const std::string& name, const point3_collection& pcs,
+               const style::fill& f, const style::stroke& s) {
+                // Create the surface
+                surface sf{};
+                sf._name = name;
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply python/src/proto.cpp:54]: No, I didn't write constructors for structs here.

**Pull request merged**